hydra-node/src/Hydra/Node/Network.hs

```diff
@@ -116,12 +116,13 @@ withNetwork ::
   IsTx tx =>
   -- | Tracer to use for logging messages.
   Tracer IO (LogEntry tx (Message tx)) ->
+  HydraVersionedProtocolNumber ->
```

Comment:
Should: not make this "configurable"

`withNetwork` here is already the concrete networking stack used by the `hydra-node` and we can either hard-code it to a given version here or in `Hydra.Network.Ouroboros`.